### ISSUE TITLE
Publish-UDDashboard succeeds but service will not start

### ISSUE THREAD

**Issue by Chirishman** [ironmansoftware/universal-dashboard#170]
Publish-UDDashboard succeeded with no errors, however the service never successfully starts.

Universal Dashboard as a Service crashes on start.

![UDError](https://user-images.githubusercontent.com/7658840/40011103-c24f3864-5774-11e8-83f1-e1fc487c8800.png)

> The PowerShell Universal Dashboard service terminated with the following error: 
> %%4294967295

**Comment by Phyziik** [comment]
Show your code. This error is given anytime you have a syntax error, like brackets and param errors. Let me see if I can see your error.

**Comment by Phyziik** [comment]
Got anything in your debug logs?

**Comment by Chirishman** [comment]
I'm actually more confused now if that's possible. I added a `Wait-Debuger` to the `dashboard.ps1` and stepped through it. When I did the dashboard ran fine if I left it paused one line past the `Start-UDDashboard` statement, however when I unpaused/detached the debugger it immediately crashed out. 

For now I'm just going to put a `Wait-Debugger` statement on line 560 because that seems to fix it.

To replicate what I'm seeing, add a wait-debugger to line one of the dashboard script, then in a different PowerShell Prompt:

```PowerShell
Enter-PSHostProcess -Id (Get-Process UniversalDashboard | select -First 1 -ExpandProperty Id)
Debug-Runspace -Runspace (Get-Runspace | ?{$_.RunspaceAvailability -eq 'InBreakpoint'})
Set-PSBreakpoint -Command Start-UDDashboard -script 'C:\Path\To\dashboard.ps1'
c
```

**Comment by adamdriscoll** [comment]
Very strange. You should be able to load external module dependencies when running as a service and I'm not quite sure why this would fail in such a way.

**Comment by Chirishman** [comment]
Yeah, the external module thing seems to be a dead end. The real issue seems to come when the process proceeds past the `Start-UDDashboard` statement in the `dashboard.ps1` file and reaches the end of the script.

If I cause it to pause after `Start-UDDashboard` but before the end of the file then everything is fine. The service is stuck forever in the state "starting" but the dashboard works in the browser.

**Comment by adamdriscoll** [comment]
@Chirishman If you put in a -Wait parameter on Start-UDDashbaord in your dashboard.ps1, does it also work? And is stuck in the Starting state?

**Comment by Chirishman** [comment]
I can test it again but I remember the -Wait parameter being unavailable in this scenario due to something about parameter sets

**Comment by adamdriscoll** [comment]
I think I reproduced your issue. The problem was that UniversalDashboard.exe was very picky about errors. If even a non-terminating error was present after creating the dashboard and starting the dashboard, the UD.exe process would terminate. I've fixed this. I've started publishing nightly builds to Azure Blob storage so if you grab the latest, it should resolve your issue: https://udnightly.blob.core.windows.net/master/1.6.2.322/Build.zip

Let me know if it works for you.
